tools/src/wyvern/target/corewyvernIL/expression/Effect.java

```diff
@@ -0,0 +1,57 @@
+package wyvern.target.corewyvernIL.expression;
+
+import java.util.HashSet;
+
+import wyvern.target.corewyvernIL.decltype.DeclType;
+import wyvern.target.corewyvernIL.decltype.EffectDeclType;
+import wyvern.target.corewyvernIL.support.TypeContext;
+import wyvern.target.corewyvernIL.type.ValueType;
+import wyvern.tools.errors.FileLocation;
+
+public class Effect {
+	Variable path;
+	String name;
+	HashSet<Effect> effectSet;
```

Comment:
This doesn't make sense; Effect shouldn't be a tree (i.e. an effect does not contain a set of effects).  Just get rid of the effectSet variable (there will still be sets of effects annotating methods and effect declarations).